x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/WaterfallItem.tsx

```diff
@@ -195,14 +200,8 @@ export function WaterfallItem({
       isSelected={isSelected}
       onClick={onClick}
     >
-      <ItemBar // using inline styles instead of props to avoid generating a css class for each item
-        style={{ left: `${left}%`, width: `${width}%` }}
-        color={color}
-        type={item.docType}
-      />
-      <ItemText // using inline styles instead of props to avoid generating a css class for each item
-        style={{ minWidth: `${Math.max(100 - left, 0)}%` }}
-      >
+      <ItemBar left={left} width={width} color={color} type={item.docType} />
+      <ItemText minWidth={Math.max(100 - left, 0)}>
```

Comment:
As counter-intuitive as it may sounds, you should use inline styles instead of props here (read the comments for details.

You could also use `attrs` but personally I don't find that easier to read: https://github.com/styled-components/styled-components/issues/134#issuecomment-312415291

Reply:
Fair enough! 👍🏻